src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -40,125 +35,70 @@
   }
 
   /**
-   * Returns a map of unit -> transport. Unit must already be loaded in the
-   * transport. If no units are loaded in the transports then an empty Map will
-   * be returned.
+   * Returns a map of unit -> transport. Tries to load units evenly across all transports.
    */
-  private static Map<Unit, Unit> mapTransportsAlreadyLoaded(final Collection<Unit> units,
+  public static Map<Unit, Unit> mapTransportsToLoad(final Collection<Unit> units,
       final Collection<Unit> transports) {
-    final Collection<Unit> canBeTransported = Match.getMatches(units, Matches.UnitCanBeTransported);
-    final Collection<Unit> canTransport = Match.getMatches(transports, Matches.UnitCanTransport);
+
+    final List<Unit> canBeTransported = sortByTransportCostDescending(units);
+    final List<Unit> canTransport = sortByTransportCapacityAscending(transports);
+
+    // Add units to transports evenly
     final Map<Unit, Unit> mapping = new HashMap<Unit, Unit>();
-    final Iterator<Unit> land = canBeTransported.iterator();
-    while (land.hasNext()) {
-      final Unit currentTransported = land.next();
-      final Unit transport = TransportTracker.transportedBy(currentTransported);
+    final IntegerMap<Unit> addedLoad = new IntegerMap<Unit>();
+    for (final Unit unit : canBeTransported) {
 
-      // already being transported, make sure it is in transports
-      if (transport == null) {
-        continue;
+      // Find first transport that has enough capacity
+      final int cost = UnitAttachment.get((unit).getType()).getTransportCost();
+      Unit loadedTransport = null;
+      for (final Unit transport : canTransport) {
+        final int capacity = TransportTracker.getAvailableCapacity(transport) - addedLoad.getInt(transport);
+        if (capacity >= cost) {
+          addedLoad.add(transport, cost);
+          mapping.put(unit, transport);
+          loadedTransport = transport;
+          break; // skip to next unit after loaded
+        }
       }
-      if (!canTransport.contains(transport)) {
-        continue;
+
+      // Move loaded transport to end of list
```

Comment:
Instead of moving the loaded transport to the end of the list, could we check if it is full, and if so, remove it?
Higher level though, even that may not be necessary. Everything here should be in-memory, so the method may run plenty fast without the optimization. I'm afraid it violated the least-surprise principle, it was curious to see an element be removed to be added again.